process-engine/bpmn-studio
# ✨ Add task type to task list

## Changes

1. Display a Task's type in the TaskList
2. Refactor TaskList component:
    - The data source now uses a customized type named `TaskListEntry`
    - The combined Task Type is now only used for internal mapping
    - Remove the `ProcessModelAnnotation` type
    - Merge all mapper functions into a generalized `mapToInternalTask` function, which produces a list of `TaskListEntry` objects
    - Remove `processModel` attachment from TaskListEntries (whatever the reason was for attaching the ProcessModel, it is no longer necessary)
    - Reorder Stuff: Public before Private and Properties before Functions
    - Make `_UpdateTasks` private
3. Patch vulnerabilities from ProcessEngineContracts and ManagementApiContracts
4. Get KPI DataModels from ManagementApiContracts
5. Remove unnecessary dependencies to KpiApiContracts, ConsumerApiContracts and ProcessEngineContracts

## Issues

Closes #1584
 
PR: #1586

## How to test the changes

- Start a couple of ProcessModels with UserTasks, ManualTasks and EmptyActivities
- Go to the Dashoard
- See that each entry in the TaskList contains the task's type
